### PR TITLE
APEXMALHAR-2529 Allow subclasses of AbstractFileInputOperator to have…

### PR DESCRIPTION
**Pull request by davidyan74**
… their own logic to advance the streaming window

@tweise Please take a look.

### CODE REVIEW

**Comment by tweise** [comment]
LGTM, will merge on Monday if there is no other feedback.

**Comment by tweise** [comment]
test this please

**Review by tweise — APPROVED** [review]

File: library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java [hunk 1]
```diff
@@ -711,6 +711,14 @@ public void emitTuples()
   }
 
   /**
+   * Whether or not to advance the window now regardless of emitBatchSize.
+   */
+  protected boolean advanceWindow()
```
Comment (tweise) [inline library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java]: The method does not "advance the window" and "window" by itself is ambiguous. Would "suspendEmit()" or something similar better convey the purpose? Maybe also expand on that in the javadoc?

Comment (davidyan74) [reply library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java]: Should it be suspendEmitTillNextWindow or something like that? I think suspendEmit can be confusing...

**Comment by davidyan74** [comment]
test this please

File: library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java (hunk 1 above)
Comment (tweise) [reply library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java]: Doesn't till when emit is suspended depend on the logic you implement?

Comment (davidyan74) [reply library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java]: You're right. Changes applied.

**Pull request merged**